docs/src/tutorials/applications/web_app.jl

```diff
@@ -0,0 +1,141 @@
+# Copyright 2017, Iain Dunning, Joey Huchette, Miles Lubin, and contributors    #src
+# This Source Code Form is subject to the terms of the Mozilla Public License   #src
+# v.2.0. If a copy of the MPL was not distributed with this file, You can       #src
+# obtain one at https://mozilla.org/MPL/2.0/.                                   #src
+
+# # Serving web apps
+
+# This tutorial demonstrates how to setup and serve JuMP models via a REST API.
+
+# First, we need the standard JuMP packages:
+
+using JuMP
+import HiGHS
+
+# We also need [HTTP.jl](https://github.com/JuliaWeb/HTTP.jl) to act as our
+# REST server, and [JSON.jl](https://github.com/JuliaIO/JSON.jl) to marshall
+# data.
+
+import HTTP
+import JSON
+
+# ## The server side
+
+# The core components of our REST server are _endpoints_. These are functions
+# which accept a `Dict{String,Any}` of input parameters, and return a
+# `Dict{String,Any}` as output. The types are `Dict{String,Any}` because we're
+# going to read these to and from JSON.
+
+# Here's a very simple endpoint: it accepts `params` as input, formulates and
+# solves a trivial mixed-integer program, and then returns a dictionary with the
+# result.
+
+function endpoint_solve(params::Dict{String,Any})
+    if !haskey(params, "lower_bound")
+        return Dict{String,Any}(
+            "status" => "failure",
+            "reason" => "missing lower_bound param",
+        )
+    end
+    model = Model(HiGHS.Optimizer)
+    set_silent(model)
+    @variable(model, x >= params["lower_bound"], Int)
+    optimize!(model)
+    return Dict{String,Any}(
+        "status" => "okay",
+        "terminaton_status" => termination_status(model),
+        "primal_status" => primal_status(model),
+        "x" => value(x),
+    )
+end
+
+# When we call this, we get:
+
+endpoint_solve(Dict{String,Any}("lower_bound" => 1.2))
+
+#-
+
+endpoint_solve(Dict{String,Any}())
+
+# For a second function, we need a function that accepts an `HTTP.Request`
+# object and returns an `HTTP.Response` object.
+
+function serve_solve(request::HTTP.Request)
+    data = JSON.parse(String(request.body))
+    solution = endpoint_solve(data)
+    return HTTP.Response(200, JSON.json(solution))
+end
+
+# Finally, we need an HTTP server. There are a variety of ways you can do this
+# in HTTP.jl. We use an explicit `Sockets.listen` so we have manual control of
+# when we shutdown the server.
+
+function setup_server(host, port)
+    server = HTTP.Sockets.listen(host, port)
+    @async HTTP.serve(host, port; server = server) do request
```

Comment:
What happens when you get multiple requests in parallel? Does the server run multiple threads?

Reply:
No. It looks like you need to write the thread handling yourself: https://github.com/JuliaWeb/HTTP.jl/issues/798